File: dot-net/src/OCP.Filter/OCP.Filter/Model/ProductFilter.cs
using System.Collections.Generic;

namespace OCP.Filter.Model
{
	public class ProductFilter
    {
        public IEnumerable<Product> ByColor(IList<Product> products, ProductColor productColor)
        {
            foreach (var product in products)
            {
                if (product.Color == productColor)
					yield return product;
            }
        }

        public IEnumerable<Product> ByColorAndSize(IList<Product> products, ProductColor productColor, ProductSize productSize)
        {
            foreach (var product in products)
            {
                if ((product.Color == productColor) && (product.Size == productSize))
					yield return product;
            }
        }

        public IEnumerable<Product> BySize(IList<Product> products, ProductSize productSize)
        {
            foreach (var product in products)
            {
                if ((product.Size == productSize))
					yield return product;
            }
        } 
    }
}

File: java/ocp-filter/src/ProductFilter.java
import java.util.ArrayList;
import java.util.List;

public class ProductFilter {
	
    public List<Product> byColor(List<Product> products, ProductColor productColor) {
    	List<Product> filtered = new ArrayList<Product>();
    	
        for(Product product: products) {
            if (product.getColor().equals(productColor))
            	filtered.add(product);
        }
        
        return filtered;
    }

    public List<Product> byColorAndSize(List<Product> products, ProductColor productColor, ProductSize productSize) {
    	List<Product> filtered = new ArrayList<Product>();
    	
        for(Product product: products) {
            if (product.getColor().equals(productColor) && product.getSize().equals(productSize))
            	filtered.add(product);
        }
        
        return filtered;
    }
    
    public List<Product> bySize(List<Product> products, ProductSize productSize) {
    	List<Product> filtered = new ArrayList<Product>();
    	
        for(Product product: products) {
            if (product.getSize().equals(productSize))
            	filtered.add(product);
        }
        
        return filtered;
    } 
}